### PR TITLE
[CARBONDATA-3927]Remove unwanted fields from tupleID to make it short and to improve store size and performance.

### PR DESCRIPTION
**Pull request by akashrn5**
### Why is this PR needed?
 Currently, we store the tupleId which is very long, which increases the store size and reduces query performance.
 
 ### What changes were proposed in this PR?
Remove the compressor name, part id and batch number from the tupleID. This helps to improve the store size and improve the query performance also.
1. part_0 is not required as it will be same for all
2. _batchno is not required as common word, can be replaced by -
3. compressor name not required as it does not plan any role.

**Before**
0/0/0-0_batchno0-0-0-1597307621800.snappy/0/0/1
**After**
0/0-0-0-0-0-1597307621800/0/0/1 

 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1617/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3358/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1632/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3373/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3378/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1637/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3501/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1759/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1762/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3504/

**Comment by kunal642** [comment]
retest this please

File: core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java [hunk 1]
```diff
@@ -952,10 +952,22 @@ public static String getSegmentBlockNameKey(String segID, String blockName,
...
+      } else {
+        return blockNameWithOutPart;
+      }
+    }
+    if (index != -1) {
+      String blockNameWithoutCompressorName =
+          blockNameWithOutPart.replace(blockNameWithOutPart.substring(index), "");
+      return segID + CarbonCommonConstants.FILE_SEPARATOR + blockNameWithoutCompressorName;
+    } else {
+      return segID + CarbonCommonConstants.FILE_SEPARATOR + blockNameWithOutPart;
```
Comment (kunal642) [inline core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java:969]: Please add a unit test to validate this functionality

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3590/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1851/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3711/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1972/

File: core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java (hunk 1 above)
Comment (akashrn5) [reply core/src/main/java/org/apache/carbondata/core/mutate/CarbonUpdateUtil.java]: done

**Comment by akashrn5** [comment]
@kunal642 added test case, please review

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3726/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1987/

**Comment by kunal642** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2010/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3751/

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**